mainwindow.cpp

```diff
@@ -64,10 +64,8 @@ MainWindow::MainWindow(QSettings &settings)
     m_ui->mainToolBar->insertWidget(m_ui->actionOpen_Password, sessionMenuButton);
     */
 
-    // TODO: Could we make connections easier?
     // We connect John process' signals with our slots.
     // John was ended.
-    // TODO: It will good to show exit status and exit code to user.
```

Comment:
Verify if JtR exits with anything different than 1 or 0. If it doesn't then there's no point in displaying exit codes.
The error message should be something along the lines of:
"John the Ripper terminated unsuccessfully. Check the Console Log for details."